### DISCUSSION TITLE
Map multiple properties to a single

### DISCUSSION THREAD

**Discussion by danielwinkler** [riok/mapperly#774] [Q&A]
Hi,

is is possible to combine multiple properties into a single?
A classic example would be 

```csharp
public class User
{
    public Guid Id { get; }
    public string Firstname { get; }
    public string Lastname { get; }
}

public class UserDto
{
    public Guid Id { get; }
    public string Name { get; } // Firstname + " " + Lastname
}
```

Thank you for any help

**Comment by latonz** [comment]
You can use [before/after](https://mapperly.riok.app/docs/configuration/before-after-map/) map for this.

**Reply by johnwnowlin** [reply]
How would I apply before or after in an IQueryable projection? Trying to add an IsDeletable property something like this.

`                IsDeletable =  a.Children.Any() == false && a.StepChildren.Any() == false
`

Where Children and StepChildren are public ICollection<ChildType>

**Reply by latonz** [reply]
This is not possible in an IQueryable projection.

**Comment by danielwinkler** [comment]
Thanks @latonz ,

that's how i solved it for now.
for the case with a **required** property:

```csharp
public class User
{
    public Guid Id { get; }
    public string Firstname { get; }
    public string Lastname { get; }
}

public class UserDto
{
    public Guid Id { get; }
    public **required** string Name { get; } // Firstname + " " + Lastname
}
```

there will be a compile time error though, which needs to be worked around by assigning something and overwriting it.
Or is there a nicer way than that?

**Reply by latonz** [reply]
It should be possible using an object factory, however with an object factory all required properties need to be initialized in the object factory. There is currently no other solution...
